### PR TITLE
feat: encode UFM findings as TOON

### PR DESCRIPTION
**Pull request by robertolopezlopez**
### Description

Encodes native UFM scan results as generic TOON through GAF's `UfmPresenter`, matching the [CLI-1838](https://snyksec.atlassian.net/browse/CLI-1838) contract. Builds on TOON MIME selection and writers from [CLI-1825](https://snyksec.atlassian.net/browse/CLI-1825).

Replaces the earlier concise SCA/Secrets projection approach with a pure-Go encoder that preserves `FindingData` JSON shape, envelope fields, and deterministic key ordering. No external TOON library.

### Changes

- Added `internal/presenters/toon/encode.go` — JSON-to-TOON encoder (tabular/list/inline forms, UTF-8 key sort, no trailing newline).
- Added `internal/presenters/toon/envelope.go` — builds `{"results":[...]}` from `TestResult.Findings()` and renders via the encoder.
- Added `internal/presenters/toon/format.go` — scalar/tabular quoting helpers aligned with TOON 4.1 rules.
- Wired `ufm.toon.tmpl` through `encodeUFMToon` and registered TOON MIME handling in `UfmPresenter` and the unified output workflow.

### Testing

- Unit golden tests for all seven CLI-1838 contract fixture pairs (`encode_test.go`).
- Presenter integration tests byte-matching contract goldens via `RenderTemplate` (`presenter_ufm_toon_test.go`).
- Output-workflow test verifies TOON routing with varied UFM fixtures.

### CLI integration

- Draft CLI PR: [snyk/cli#7241](https://github.com/snyk/cli/pull/7241) — pins `cliv2` to `c41b258` on this branch (`v0.22.2-0.20260908104610-c41b258683da`).

Closes [CLI-1826](https://snyksec.atlassian.net/browse/CLI-1826).

### Checklist

- [x] Tests added and all succeed (`make test`)
- [x] Regenerated mocks, etc. (`make generate`)
- [x] Linted (`make lint`)
- [ ] Test your changes work for the CLI
  1. Clone / pull the latest [CLI](https://github.com/snyk/cli) main.
  2. Run `go get github.com/snyk/go-application-framework@YOUR_LATEST_GAF_COMMIT` in the `cliv2` directory.
      - Tip: for local testing, you can uncomment the line near the bottom of the CLI's [`go.mod`](https://github.com/snyk/cli/blob/main/cliv2/go.mod) to point to your local GAF code.
  3. Run `go mod tidy` in the `cliv2` directory.
  4. Run the CLI tests and do any required manual testing.
  5. Draft CLI integration PR: [snyk/cli#7241](https://github.com/snyk/cli/pull/7241) (re-pin to post-merge GAF main before merging).
  - Once this PR is merged, repeat these steps, but pointing to the latest GAF commit on main and update your CLI PR.

[CLI-1838]: https://snyksec.atlassian.net/browse/CLI-1838?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CLI-1825]: https://snyksec.atlassian.net/browse/CLI-1825?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CLI-1826]: https://snyksec.atlassian.net/browse/CLI-1826?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

> [!NOTE]
> **Medium Risk**
> New CLI-facing serialization for full UFM payloads; mistakes could break consumers or misrepresent findings, though scope is presentation-only and heavily golden-tested.
> 
> **Overview**
> Adds a **TOON output path** for unified findings model (UFM) scan results, alongside existing human, SARIF, and HTML formats.
> 
> A new **`internal/presenters/toon`** package builds a `{"results":[...]}` envelope from `TestResult` (including `Findings()`), normalizes via JSON round-trip with `UseNumber`, and supplies template helpers for tabular arrays, quoting (TOON 4.1-style), and key formatting. **`ufm.toon.tmpl`** walks that structure recursively; **`UfmPresenter`** registers `ApplicationTOONMimeType` with `getToonTemplateFuncMap`, and the **unified output workflow** maps the TOON MIME type to `ApplicationTOONTemplatesUfm`.
> 
> Tests add contract golden fixtures, presenter byte-matching, edge cases (findings errors, non-JSON-safe values, control characters), and a workflow check that the TOON config flag selects the TOON writer.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c41b258683da2738e3366a427c5eb3a7701ac243. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/presenters/templates/ufm.toon.tmpl [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{{- define "main" -}}
+{{- $types := getFindingTypesFromMultipleTestResults .TestResults -}}
+{{- $wroteSCA := false -}}
+{{- if containsFindingType $types "sca" -}}
+{{- $sca := projectSCA .TestResults -}}
+{{- if gt (len $sca.Rows) 0 -}}
+sca[{{len $sca.Rows}}]{id,severity,pkg,fixable}:
```
Comment (PeterSchafer) [inline internal/presenters/templates/ufm.toon.tmpl]: question: why don't we dynamically derive the output from findingstype instead of hardcoding certain types? also below for secrets. the templates should be agnostic as much as possible. New products should just work.

File: internal/presenters/toon/sca.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	"strconv"
+	"strings"
+
+	"github.com/snyk/go-application-framework/pkg/apiclients/testapi"
+)
+
+var scaSeverityOrder = []string{"critical", "high", "medium", "low"}
+
+// ProjectSCA maps native UFM SCA findings to the concise TOON view model.
+func ProjectSCA(results []testapi.TestResult) (SCAView, error) {
```
Comment (octavian-snyk) [inline internal/presenters/toon/sca.go]: Can we create something generic for UFM findings instead of building separate implementations for each product?

File: internal/presenters/templates/ufm.toon.tmpl [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{{- define "main" -}}
+{{- encodeUFMToon .TestResults -}}
```
Comment (PeterSchafer) [inline internal/presenters/templates/ufm.toon.tmpl]: Issue: this is basically bypassing the template based rendering

Comment (robertolopezlopez) [reply internal/presenters/templates/ufm.toon.tmpl]: hi, I have just seen this message now. Yes that is the case - template should own the presentation logic. I changed that yesterday evening

**Review by octavian-snyk — APPROVED** [review]
Looks good as a first iteration.
Changes are isolated to TOON rendering.

⚠️ Associated CLI PR must pass tests before merging. ⚠️
